_source/_posts/2021-10-16-full-stack-java.md

```diff
@@ -0,0 +1,691 @@
+```
+
+You can also remove all the getters and setters for these fields in `src/test/javascript/e2e/entities/photo/photo-update.page-object.ts`:
+
+```js
+setHeightInput(height) {
+  this.heightInput.sendKeys(height);
+}
+
+getHeightInput() {
+  return this.heightInput.getAttribute('value');
+}
+
+setWidthInput(width) {
+  this.widthInput.sendKeys(width);
+}
+
+getWidthInput() {
+  return this.widthInput.getAttribute('value');
+}
+
+setTakenInput(taken) {
+  this.takenInput.sendKeys(taken);
+}
+
+getTakenInput() {
+  return this.takenInput.getAttribute('value');
+}
+
+setUploadedInput(uploaded) {
+  this.uploadedInput.sendKeys(uploaded);
+}
+
+getUploadedInput() {
+  return this.uploadedInput.getAttribute('value');
+}
+```
+
+Stop your Maven process, run `npm webapp:build`, start Maven again, and run `npm run e2e` to make sure everything still works. 
+
+**NOTE**: If you experience authentication errors in your Cypress tests, it's likely because you've violated Auth0's [Rate Limit Policy](https://auth0.com/docs/support/policies/rate-limit-policy). As a workaround, I recommend you use Keycloak for Cypress tests. You can do this by opening a new terminal window and starting your app there (without running `source .auth0.env`).
+
+If you upload an image you took with your smartphone, the height, width, and taken values should all be populated. If they're not, chances are your image doesn't have the data in it.
+
+> Need some sample photos with EXIF data? You can download pictures of my 1966 VW Bus from [an album on Flickr](https://www.flickr.com/photos/mraible/albums/72157689027458320).
+
+## Add a React Photo Gallery
+
+You've added metadata extraction to your backend, but your photos still display in a list rather than in a grid (like Flickr). To fix that, you can use [React Photo Gallery](https://github.com/neptunian/react-photo-gallery) component. Install it using npm:
+
+```bash
+npm i react-photo-gallery@8
+```
+
+In `src/main/webapp/app/entities/photo/photo.tsx`, add an import for `Gallery`:
+
+```ts
+import Gallery from 'react-photo-gallery';
+```
+
+Then add a `photoSet` variable in the `render()` method, and the `<Gallery>` component right after the closing `</h2>`.
+
+```ts
+render() {
+  const { photoList, match } = this.props;
+  const photoSet = photoList.map(photo => ({
+    src: `data:${photo.imageContentType};base64,${photo.image}`,
+    width: photo.height > photo.width ? 3 : photo.height === photo.width ? 1 : 4,
+    height: photo.height > photo.width ? 4 : photo.height === photo.width ? 1 : 3
+  }));
+
+  return (
+    <div>
+      <h2 id="photo-heading">
+        ...
+      </h2>
+      <Gallery photos={photoSet} />
+      ...
+  );
+}
+```
+
+Since you only modified the front end code, you can run `npm start` to start an instance of webpack-dev-server that proxies requests to the backend and auto-refreshes your browser (using [Browsersync](https://browsersync.io/)) every time you change any React files.
```

Comment:
```suggestion
Since you only modified the front-end code, you can run `npm start` to start an instance of webpack-dev-server that proxies requests to the backend and auto-refreshes your browser (using [Browsersync](https://browsersync.io/)) every time you change any React files.
```
a fussy hypen for 'front-end' used as a modifier